### PR TITLE
Timeout option

### PR DESCRIPTION
**Pull request by noma4i**
Added Timeout option for Exec block. This will prevent powershell script from dying by timeout. 
From now default timeout is extended to 1800 seconds and could be passed as an argument to download_file block.

````puppet
download_file { "Download File" :
  url => "http://slow_website/extra_big_file.zip",
  destination_directory => 'c:\\Temp',
  destination_file => 'big_and_slow.zip',
  timeout => 90000
}
````

### CODE REVIEW

**Comment by noma4i** [comment]
Build fail is not related to my changes. As I see thats issue with gems.

**Comment by ffrank** [comment]
Hi, sorry for the delay.

I feel that the default should either be what is currently implicitly used, or a somewhat more sensible value. 30 minutes seem excessive. Most users will not want to wait that long in case something is stuck during execution.

**Comment by noma4i** [comment]
Sure, maybe 3m default is better

**Comment by daenney** [comment]
@noma4i Could you please update the PR?

File: manifests/init.pp [hunk 1]
```diff
@@ -61,30 +66,25 @@
...
-  if defined(File['C:\temp']) {
-    $tmp_dir = 'C:\temp'
-  } elsif defined(File['C:/temp']) {
-    $tmp_dir = 'C:/temp'
-  } else {
-    $tmp_dir = 'C:\temp'
-
-    file { $tmp_dir:
+  if ! defined(File['C:\temp']) {
+    file { 'C:\temp':
```
Comment (ffrank) [inline manifests/init.pp]: Why are you changing this?

I feel that this hurts the interoperability of this module a bit.

**Comment by ffrank** [comment]
To move this forward, I made a suggestion in PR #23. The feature is still attributable to you of course.

**Pull request closed without merging**